lib/src/convert.dart

```diff
@@ -273,15 +275,23 @@ class StringDecoder extends DataDecoder<String> {
   const StringDecoder();
 
   @override
-  String convert(List<int> input) => utf8.decode(input);
+  String convert(List<int> input) {
+    final nextNull = input.indexOf(0);
+    if (nextNull == -1) {
```

Comment:
Nice!

Would be great to get a test that validates this?

Reply:
I'm not good at designing tests... I've never done testing in dart or really in any language. Sorry to drop the ball on that.

Reply:
No worries.

I'm happy to help.  If it's easy, maybe give me an example where the previous implementation was falling down?  (What strings were you decoding that had `null`s?)